### ISSUE TITLE
Not Flashing Properly on OSX

### ISSUE THREAD

**Issue by Rabitzwin** [trezor/trezor-core#246]
I know I'm probably just missing a simple step somewhere, but I followed the hardware build guide and my "dev model" trezor (built with the microcontroller on the fancy testing board, the capacitive touch module/screen, and a micro sd reader with lots of soft wires) is not functioning. As far as I can tell I'm recieving proper feedback from the flashing command (included below), yet nothing appears on the screen and the trezor commands that I've been attempting to use from the library do not recognize this as a trezor device.

Any idea what could be wrong? Any and all feedback is appreciated (I apologize if this is the wrong way to ask for help, I didn't know how else to do it)

> openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/boardloader/boardloader.bin 0x08000000; exit"
Open On-Chip Debugger 0.10.0
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v25 API v2 SWIM v14 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.875485
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread
xPSR: 0x01000000 pc: 0x08004b84 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread
xPSR: 0x61000000 pc: 0x20000046 msp: 0x10010000
wrote 49152 bytes from file build/boardloader/boardloader.bin in 1.698892s (28.254 KiB/s)
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/bootloader/bootloader.bin 0x08020000; exit"
Open On-Chip Debugger 0.10.0
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v25 API v2 SWIM v14 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.875485
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread
xPSR: 0x01000000 pc: 0x08004b84 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread
xPSR: 0x61000000 pc: 0x20000046 msp: 0x10010000
wrote 131072 bytes from file build/bootloader/bootloader.bin in 3.565372s (35.901 KiB/s)
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/firmware/firmware.bin 0x08040000; exit"
Open On-Chip Debugger 0.10.0
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v25 API v2 SWIM v14 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.875485
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread
xPSR: 0x01000000 pc: 0x08004b84 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread
xPSR: 0x61000000 pc: 0x20000046 msp: 0x10010000
wrote 786432 bytes from file build/firmware/firmware.bin in 21.341629s (35.986 KiB/s)
potatochips:trezor-core Eswift125$ c "init; reset halt; flash write_image erase build/boardloader/boardloader.bin 0x08000000; exit"
-bash: c: command not found
potatochips:trezor-core Eswift125$ Open On-Chip Debugger 0.10.0
The files /Users/Eswift125/trezor-core/On-Chip, /Users/Eswift125/trezor-core/Debugger, and /Users/Eswift125/trezor-core/0.10.0 do not exist.
potatochips:trezor-core Eswift125$ Licensed under GNU GPL v2
-bash: Licensed: command not found
potatochips:trezor-core Eswift125$ For bug reports, read
-bash: For: command not found
potatochips:trezor-core Eswift125$ http://openocd.org/doc/doxygen/bugs.html
-bash: http://openocd.org/doc/doxygen/bugs.html: No such file or directory
potatochips:trezor-core Eswift125$ hla_swd
-bash: hla_swd: command not found
potatochips:trezor-core Eswift125$ Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v25 API v2 SWIM v14 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.875485
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread
xPSR: 0x01000000 pc: 0x08004b84 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread
xPSR: 0x61000000 pc: 0x20000046 msp: 0x10010000
wrote 49152 bytes from file build/boardloader/boardloader.bin in 1.698892s (28.254 KiB/s)

**Comment by prusnak** [comment]
It's the right way to ask for help :-)

what board do you use?

**Comment by Rabitzwin** [comment]
http://www.st.com/en/evaluation-tools/32f429idiscovery.html

**Comment by prusnak** [comment]
I guess there is something broken with your script.

There is `-bash: hla_swd: command not found` and similar in your output.

How exactly do you flash the device?

**Comment by Rabitzwin** [comment]
That's exactly what I used, although I was using the STLINK interface so I had to set the version when I used it (as specified in the instructions, although I eventually got tired of doing that and changed the version in the makefile). I believe I installed all things specified in the instructions, but its possible something got messed up.

**Comment by mcudev** [comment]
hi guys, sorry for not putting together a tutorial page like i did for v1 yet. i mean to.

fortunately the dev board, build, and flash stuff still works for me, so i can try to help.

the OP's output looks ok up until the line that contains "potato" (writing the boardloader, bootloader, and firmware looks like it worked). not sure what's going on starting at that point in the output.

i just build with this single command line:
`pipenv run make clean vendor build_boardloader build_bootloader build_firmware STLINK_VER=v2-1 DISPLAY_ILI9341V=1`

i just flashed with these single command lines:

`make flash STLINK_VER=v2-1`
or
`make flash openocd_reset STLINK_VER=v2-1`

the second way of flashing is for when i'm too lazy to reach over and press the black reset button on the dev board. you only need to run one of them.

i'm using Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (git commit 4999c9d98013eeb9746082f6b018e4b763ae3d37 from git://git.code.sf.net/p/openocd/code.git)

here's what success looks like to me:
```
trezor-core$ make flash STLINK_VER=v2-1
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/boardloader/boardloader.bin 0x08000000; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.869321
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08004ca4 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread 
xPSR: 0x61000000 pc: 0x20000044 msp: 0x10010000
wrote 49152 bytes from file build/boardloader/boardloader.bin in 1.574255s (30.491 KiB/s)
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/bootloader/bootloader.bin 0x08020000; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.867916
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08004ca4 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread 
xPSR: 0x61000000 pc: 0x20000044 msp: 0x10010000
wrote 131072 bytes from file build/bootloader/bootloader.bin in 3.239056s (39.518 KiB/s)
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/firmware/firmware.bin 0x08040000; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.867916
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08004ca4 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread 
xPSR: 0x61000000 pc: 0x20000044 msp: 0x10010000
wrote 1179648 bytes from file build/firmware/firmware.bin in 27.970549s (41.186 KiB/s)

trezor-core$ make flash openocd_reset STLINK_VER=v2-1
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/boardloader/boardloader.bin 0x08000000; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.869321
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08004ca4 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread 
xPSR: 0x61000000 pc: 0x20000044 msp: 0x10010000
wrote 49152 bytes from file build/boardloader/boardloader.bin in 1.568411s (30.604 KiB/s)
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/bootloader/bootloader.bin 0x08020000; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.867916
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08004ca4 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread 
xPSR: 0x61000000 pc: 0x20000044 msp: 0x10010000
wrote 131072 bytes from file build/bootloader/bootloader.bin in 3.358437s (38.113 KiB/s)
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash write_image erase build/firmware/firmware.bin 0x08040000; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.869321
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08004ca4 msp: 0x10010000
auto erase enabled
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
target halted due to breakpoint, current mode: Thread 
xPSR: 0x61000000 pc: 0x20000044 msp: 0x10010000
wrote 1179648 bytes from file build/firmware/firmware.bin in 27.948088s (41.219 KiB/s)
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.869321
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz

trezor-core$ make flash_erase STLINK_VER=v2-1
openocd -f interface/stlink-v2-1.cfg -c "transport select hla_swd" -f target/stm32f4x.cfg -c "init; reset halt; flash info 0; flash erase_sector 0 0 last; flash erase_check 0; exit"
Open On-Chip Debugger 0.10.0+dev-00449-g4999c9d9 (2018-07-04-12:39)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2-1.cfg is deprecated, please switch to interface/stlink.cfg
hla_swd
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
adapter speed: 2000 kHz
adapter_nsrst_delay: 100
none separate
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : clock speed 1800 kHz
Info : STLINK v2 JTAG v30 API v2 SWIM v19 VID 0x0483 PID 0x374B
Info : using stlink api v2
Info : Target voltage: 2.867916
Info : stm32f4x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : Listening on port 3333 for gdb connections
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x08004ca4 msp: 0x10010000
Info : device id = 0x20016419
Info : flash size = 2048kbytes
Info : Dual Bank 2048 kiB STM32F42x/43x/469/479 found
#0 : stm32f2x at 0x08000000, size 0x00200000, buswidth 0, chipwidth 0
	#  0: 0x00000000 (0x4000 16kB) not protected
	#  1: 0x00004000 (0x4000 16kB) not protected
	#  2: 0x00008000 (0x4000 16kB) not protected
	#  3: 0x0000c000 (0x4000 16kB) not protected
	#  4: 0x00010000 (0x10000 64kB) not protected
	#  5: 0x00020000 (0x20000 128kB) not protected
	#  6: 0x00040000 (0x20000 128kB) not protected
	#  7: 0x00060000 (0x20000 128kB) not protected
	#  8: 0x00080000 (0x20000 128kB) not protected
	#  9: 0x000a0000 (0x20000 128kB) not protected
	# 10: 0x000c0000 (0x20000 128kB) not protected
	# 11: 0x000e0000 (0x20000 128kB) not protected
	# 12: 0x00100000 (0x4000 16kB) not protected
	# 13: 0x00104000 (0x4000 16kB) not protected
	# 14: 0x00108000 (0x4000 16kB) not protected
	# 15: 0x0010c000 (0x4000 16kB) not protected
	# 16: 0x00110000 (0x10000 64kB) not protected
	# 17: 0x00120000 (0x20000 128kB) not protected
	# 18: 0x00140000 (0x20000 128kB) not protected
	# 19: 0x00160000 (0x20000 128kB) not protected
	# 20: 0x00180000 (0x20000 128kB) not protected
	# 21: 0x001a0000 (0x20000 128kB) not protected
	# 22: 0x001c0000 (0x20000 128kB) not protected
	# 23: 0x001e0000 (0x20000 128kB) not protected
STM32F4xx - Rev: 3
erased sectors 0 through 23 on flash bank 0 in 30.825680s
target halted due to breakpoint, current mode: Thread 
xPSR: 0x41000000 pc: 0x20000022 msp: 0x10010000
successfully checked erase state
	Bank is erased
```

**Comment by mcudev** [comment]
Also, this patch helps me debug and use the firmware that i build when using the dev board:
```
trezor-core$ cat ../model-t-developer.patch
diff --git a/SConscript.boardloader b/SConscript.boardloader
index 4bd4afc7..4a52e4b1 100644
--- a/SConscript.boardloader
+++ b/SConscript.boardloader
@@ -99,7 +99,7 @@ env.Replace(
     OBJCOPY='arm-none-eabi-objcopy', )
 
 env.Replace(
-    CCFLAGS='-Os '
+    CCFLAGS='-Og '
     '-g3 '
     '-nostdlib '
     '-std=gnu99 -Wall -Werror -Wdouble-promotion -Wpointer-arith -fno-common '
diff --git a/SConscript.bootloader b/SConscript.bootloader
index 4f808a99..26422f69 100644
--- a/SConscript.bootloader
+++ b/SConscript.bootloader
@@ -120,7 +120,7 @@ env.Replace(
     OBJCOPY='arm-none-eabi-objcopy', )
 
 env.Replace(
-    CCFLAGS='-Os '
+    CCFLAGS='-Og '
     '-g3 '
     '-nostdlib '
     '-std=gnu99 -Wall -Werror -Wdouble-promotion -Wpointer-arith -fno-common '
diff --git a/SConscript.firmware b/SConscript.firmware
index f8ad202b..dbfcee80 100644
--- a/SConscript.firmware
+++ b/SConscript.firmware
@@ -309,7 +309,7 @@ env.Replace(
     OBJCOPY='arm-none-eabi-objcopy', )
 
 env.Replace(
-    COPT='-Os',
+    COPT='-Og',
     CCFLAGS='$COPT '
     '-g3 '
     '-nostdlib '
@@ -402,7 +402,7 @@ obj_program.extend(env.Object(source=SOURCE_STMHAL))
 obj_program.extend(env.Object(source=SOURCE_TREZORHAL))
 obj_program.extend(env.Object(source=source_mpyc))
 
-VENDORHEADER = 'embed/vendorheader/vendorheader_' + ('unsafe_signed_prod.bin' if ARGUMENTS.get('PRODUCTION', '0') == '0' else 'satoshilabs_signed_prod.bin')
+VENDORHEADER = 'embed/vendorheader/vendorheader_' + ('unsafe_signed_dev.bin' if ARGUMENTS.get('PRODUCTION', '0') == '0' else 'satoshilabs_signed_prod.bin')
 
 obj_program.extend(
     env.Command(
```
use it like this:
`trezor-core$ git apply ../model-t-developer.patch`

then rebuild the firmware and flash it to the device like this:
`python-trezor$ ./trezorctl firmware_update -f ../trezor-core/build/firmware/firmware.bin`

**Comment by mcudev** [comment]
@Rabitzwin I just remembered that I made some physical modifications to my dev kit board (the STM32F429I-DISC1). See https://github.com/trezor/trezor-core/pull/40#issuecomment-336086174. Right now, I have removed the external SDRAM chip and the original display that came on the board. IIRC, the display can be torn off pretty easily and the SDRAM chip was easier to remove with an X-Acto knife than by de-soldering. If you decide to make these mods, be careful. It's easy to slip and slice yourself by accident. I forget if removing only one of the two components is necessary. It definitely works with both removed. This is something I should figure out and then write up in a tutorial.
